core/templates/pages/exploration-editor-page/feedback-tab/services/thread-data-backend-api.service.ts

```diff
@@ -226,8 +237,9 @@ export class ThreadDataBackendApiService {
     });
   }
 
+  // A thread will be null if threadId is invalid.
```

Comment:
I'm seeing this pattern a lot in this file, can we reduce the number of places where `null` is returned?

Reply:
This comes from Line 148 
![image](https://user-images.githubusercontent.com/60519303/128311739-b2153a4c-035d-4370-9d17-138ece3a1094.png).
Instead of throwing error at this function(getter) we're throwing errors at each setter(inside async functions). So i think `null` should be a valid value. Please note that null is not being returned in these functions, it is only a valid type for arguments provided.

Reply:
Hmm, so can we move the error in the getter?

Reply:
We could, but we generally don't throw error at getter so that error in a sense propagates more and is thrown at the right place. Please take a look at conversation done here https://github.com/oppia/oppia/pull/13204#pullrequestreview-695481342 regarding this.

Reply:
Hmmm, okay, but then can we make sure that in the places where we set `threadsById`, it is not missing the thread IDs?

Reply:
Yes it is used currently at two places in feeback-thread component and we're throwing errors at both places so fully catered.
![image](https://user-images.githubusercontent.com/60519303/128774010-176ad7cb-c8f0-4901-91b2-54fd6cae2f14.png)
![image](https://user-images.githubusercontent.com/60519303/128774047-3b5754ef-f8ae-4d0f-9d92-06ddc8bb4122.png)

Reply:
But then why do we still need to have the `null` here? If we make sure in all callsites of `markThreadAsSeenAsync` that the thread is not `null` we can disallow it here, right?